lib/davinci_dtr_test_kit/client_groups/adaptive_questionnaire/custom/dtr_full_ehr_custom_adaptive_workflow_group.rb

```diff
@@ -0,0 +1,128 @@
+      Testers will provide a custom adaptive Questionnaire package for the test, along with a list
+      (JSON array) of Questionnaires to be included in each `$next-question` response.
+
+      As part of this workflow, the Full EHR system must demonstrate its ability to:
+
+      1. Request the Questionnaire using the `$questionnaire-package` operation.
+      2. Support the tester in completing the questionnaire through multiple `$next-question` interactions, including:
+         - Rendering the questionnaire.
+         - Pre-populating answers into the questionnaire.
+         - Allowing the tester to manually enter responses, including overriding pre-populated answers.
+      3. Complete and store the `QuestionnaireResponse` for future use.
+
+      Inferno will process `$next-question` requests dynamically:
+      - Each request will receive the next Questionnaire from the provided list.
+      - If a `$next-question` request is received when the list is empty, Inferno will mark
+        the `QuestionnaireResponse` as completed.
+
+      At least two answers should be pre-populated across all sets of questions.
+    )
+    run_as_group
+    config(
+      options: { form_type: 'adaptive', next_tag: "custom_#{CLIENT_NEXT_TAG}" },
+      inputs: {
+        custom_questionnaire_package_response: {
+          name: 'adaptive_custom_questionnaire_package_response',
+          title: 'Custom Questionnaire Package Response JSON for Adaptive form',
+          description: %(
+              A JSON PackageBundle may be provided here to replace Inferno's response to
+              the $questionnaire-package request. Note: Ensure that the questionnaire package
+              has an empty Adaptive Questionnaire.
+            )
+        }
+      }
+    )
+
+    input_order :access_token
+
+    group do
+      id :dtr_full_ehr_custom_adaptive_retrieval
+      title 'Retrieving the Adaptive Questionnaire'
+
+      # Test 0: attest to launch
+      test from: :dtr_full_ehr_launch_attest,
+           config: {
+             options: {
+               attestation_message: 'I attest that DTR has been launched in the context of a patient with data that will exercise pre-population logic in the provided static questionnaire resulting in at least 2 pre-populated answers.' # rubocop:disable Layout/LineLength
+             }
+           },
+           title: 'Launch DTR (Attestation)'
+      # Test 1: Recieve questionnaire-package and next-question requests
+      test from: :dtr_full_ehr_custom_adative_request
+      # Test 2: validate the $questionnaire-package request body
+      test from: :dtr_qp_request_validation
+      # Test 3: validate the $next-question requests body
+      test from: :dtr_adaptive_next_question_request_validation
+      # Test 4: validate the QuestionnaireResponse in the input parameter
+      test from: :dtr_adaptive_response_validation do
```

Comment:
I get a purple error when I make more $next-question requests than in the input:
<img width="1229" alt="image" src="https://github.com/user-attachments/assets/f96ced24-ec43-4b15-8d6a-830d164f3515" />

Reply:
fixed